src/main/resources/static/js/shareToggleButton.js

```diff
@@ -0,0 +1,28 @@
+class ToggleButton {
+    constructor(toggleId, sendUrl) {
+        this.toggleButton = document.getElementById(toggleId);
+        this.sendUrl = sendUrl;
+
+        if (!this.toggleButton) return;
+
+        this.initialize();
+    }
+
+    initialize() {
+        this.toggleButton.addEventListener('change', () => this.toggleChange());
+    }
+
+    async toggleChange() {
+        try{
+            const isPublic = this.toggleButton.checked;
+            const response = await fetch(`${this.sendUrl}?is_public=${isPublic}`);
```

Comment:
```suggestion
            await fetch(`${this.sendUrl}?is_public=${isPublic}`);
```
response使ってないからこれでもよさそう